### ISSUE TITLE
Add integration test

### ISSUE THREAD

**Issue by dorukkilitcioglu** [nickgreenquist/cu2rec#3]
When the proper errors and low enough learning rate are given, training loss should decrease with each iteration of the training loop.

**Comment by dorukkilitcioglu** [comment]
Added training loop in the training branch.
```
[dk3306@gpu-33 tests]$ bin/test_training
Testing training loop...
Losses: 161.9 160.99 160.988 160.076 160.072 159.159 159.153 158.238 158.23 157.314
```
Looks like the loss is decreasing!

Testing for correctness will come later.
